packages/mgt-components/src/components/mgt-people/mgt-people.ts

```diff
@@ -449,6 +449,8 @@ export class MgtPeople extends MgtTemplatedComponent {
           this.people = await getPeople(graph);
         }
 
+        this.people = this.people.filter(person => person !== null && person !== '');
```

Comment:
this.people is IDynmaicPerson[] so I'm a little confused as to how person could be an empty string here.

Reply:
I'm not certain about this however, when you do a first cache of people and there is no data returned, then you end up caching empty values with a `cacheKey` derived from the default cache key values. On the next run you will hit the cache and will need to check for empty strings.

Reply:
Hmm that sounds like there might be a bug in the caching layer if we're returning strings instead of objects in some instances.

If you can work up some repro steps and raise a separate issue that would be much appreciated.

Reply:
@gavinbarron I drilled a bit into this and found the problem. In the `mgt-agenda` component, `mgt-people` is used with the `people-queries` property. This calls the `getUsersForPeopleQueries` function. Within that function, you successfully cache the users on the first run, and on the consequent runs, you fetch the users from the cache, as well as make calls to the graph. When you make the call to graph, the batch has no requests set as they are skipped cos the cache results are available. Here is where the `null` people results are set, and you get them in the people array. The filter on people fixes this and we can ignore it but I have made fix to check if the batch has requests and only perform that calls when required. You can see the agenda component loads a little faster after that.